### DISCUSSION TITLE
Security tests: No auth

### DISCUSSION THREAD

**Discussion by savage-alex** [apideck-libraries/portman#223] [Q&A] [answered]
**User Story**
As an API tester
I want to have a test that calls each API operation with no security
So that any miss configured APIs without security are found automatically

**Acceptance Criteria**
* New configuration exists to enable feature
* New folder for path + (No Authentication) exists when imported to Postman
* Calls are made with authorization set to No Auth
* Response codes and schema validation match the 401 or 403 schemas

**Comment by nicklloyd** [comment]
Hey Alex - have you looked into variation testing?
ie: https://github.com/apideck-libraries/portman/blob/main/examples/testsuite-variation-tests/portman-config.crm.json#L33-L57

**Comment by savage-alex** [comment]
I had read the docs but failed to notice the examples here! I will setup some scenarios thanks!

**Comment by nicklloyd** [comment, marked as answer]
Indeed having a dedicated (sexy) docs site is on the list, just haven't found the bandwidth to get it done just yet... 

Maybe to note, a bad token is not the same as no auth headers.... can use `overwriteRequestHeaders` to actually remove the authorization entirely...

https://github.com/apideck-libraries/portman#portman---overwrites-properties

```
{
  "overwrites": [
    {
      "openApiOperationId": "leadsAdd",
      "overwriteRequestHeaders": [
        {
          "key": "x-apideck-consumer-id",
          "value": "portman-id-{{$randomInt}}",
          "remove": true
        }
      ]
    }
  ]
}
```

**Reply by savage-alex** [reply]
Testing Postman if the security is set to bearer and you leave the token blank it results in No Authorization header being sent! result 
![image](https://user-images.githubusercontent.com/25430683/146983428-0c3fb5dd-5763-4463-93df-8a0f7eefd049.png)
![image](https://user-images.githubusercontent.com/25430683/146983854-ebdb5249-b8dc-4797-8b2a-fb49ebfd7919.png)

**Reply by savage-alex** [reply]
NB: If you enter anything into the token the header is then included 
![image](https://user-images.githubusercontent.com/25430683/146983717-862fe6fd-2b1f-461e-ac69-3c7cc8d9a998.png)
